gpflow/optimizers/mcmc.py

```diff
@@ -84,6 +84,13 @@ def _target_log_prob_fn_closure(*variables):
             with tf.GradientTape(watch_accessed_variables=False) as tape:
                 tape.watch(variables_list)
                 log_prob = self._target_log_prob_fn()
+                # Now need to correct for the fact that the prob fn is evaluated on the
+                # constrained space while we wish to evaluate it in the unconstrained space
+                for param in self._model_parameters:
+                    if isinstance(param, Parameter) and param.transform is not None:
+                        value = param.read_value()
```

Comment:
should this not be the *un*constrained version here ?
Also, should this have a check on whether param.prior_on is CONSTRAINED or UNCONSTRAINED?

Reply:
Yep thanks, fixed! I don't think it needs to check param.prior_on because evaluation of the prior is always on the constrained variable, so a correction always needs to be applied.

Reply:
Right- the point is that when prior is on unconstrained then the correction here will simply cancel the one that is applied in `log_prior`?